napari/layers/base/base.py

```diff
@@ -1403,6 +1403,11 @@ def get_ray_intersections(
 
         # create the bounding box in data coordinates
         bounding_box = self._display_bounding_box(dims_displayed)
+        # Since coordinate 0 corresponds to a pixel center, in order to
+        # correctly get the data value under the mouse at the pixel
+        # edges, we need to shift by 0.5 pixels on each axis here.
+        if not world:
+            position = tuple(p + 0.5 for p in position)
```

Comment:
@grlee77 can you clarify the comment a bit further? e.g. "since our world coordinate conventions correspond to integer values being at the pixel centers, *and VisPy instead places integer values at pixel corners*, ..." Is that accurate?

Reply:
Yes, I think that is correct, but I don't have an immediate reference for that on the VisPy side. The docs do state that (0,0) is the upper left corner of the canvas, so I think this means the "corner" of the pixel while we have (0,0) as the center of the upper left pixel.

Reply:
[This line](https://github.com/napari/napari/blob/1ba738c36d7ac0faf9bc7b5e8a2228cd93e656f2/napari/_qt/qt_viewer.py#L895) seems to be where the position gets set. That method on the QtViewer class is using the VisPy scene transforms to map the coordinate for mouse events.